### ISSUE TITLE
Scrub not fully applied on HTML::Document

### ISSUE THREAD

**Issue by chengguangnan** [flavorjones/loofah#80]
I noticed that some HTML comment tags are not removed. 

Here is an example, `my_scrub` should remove all the comments.

```
Loofah.document("<!DOCTYPE html><!--[if IE 7]><!-- --><html><body><script></script></body></html><!--ww -->").scrub!(my_scrub).to_xml
=> "<!DOCTYPE html>\n<!--[if IE 7]><!-- --><html></html>\n"
```

I check the code and think the problem is here:

https://github.com/flavorjones/loofah/blob/master/lib/loofah/instance_methods.rb#L41

```
        case self
        when Nokogiri::XML::Document
          scrubber.traverse(root) if root
        when Nokogiri::XML::DocumentFragment
          children.scrub! scrubber
        else
          scrubber.traverse(self)
        end
```

So even a `HTML::Document` would went through `scrubber.traverse(root) if root`. So things outside of HTML will not went through this scrubber.

**Comment by flavorjones** [comment]
Apologies for the atrociously long delay in responding. I understand what you're reporting, and acknowledge that the comments outside of the `html` tag are not being scrubbed.

**Comment by flavorjones** [comment]
Fix will be in v2.5.0.

Note: in the fix I've chosen to _remove_ any comments from a `Loofah::HTML::Document` that exist outside the `html` tag. I could have applied the scrubber to non-`html` root nodes, but unfortunately these nodes (or rather, the document itself) don't meet the contract expected by the scrubber (for example, these comments can't be replaced by an arbitrary node type, or have a sibling node added).
